soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -93,12 +112,92 @@ pub struct Token;
 
 #[contractimpl]
 impl TokenTrait for Token {
-    fn initialize(e: &Host, admin: Identifier, metadata: Metadata) -> Result<(), Error> {
-        if has_administrator(&e)? {
+    fn init_wrap(e: &Host, metadata: ClassicMetadata) -> Result<(), Error> {
+        if has_metadata(&e)? {
             return Err(Error::ContractError);
         }
+
+        match metadata.clone() {
+            ClassicMetadata::Native => {
+                let contract_id = BytesN::<32>::try_from_val(e, e.get_current_contract()?)?;
```

Comment:
This can be moved out of match, can't it?

Reply:
Good catch. This was a result of a couple rounds of experimenting. Fixed.